src/mongocxx/exception/authentication_exception.hpp

```diff
@@ -16,50 +16,43 @@
 
 #include <mongocxx/config/prelude.hpp>
 
-#include <cstdint>
-#include <exception>
-#include <tuple>
-
 #include <bsoncxx/document/value.hpp>
-#include <bsoncxx/document/view.hpp>
 #include <bsoncxx/stdx/optional.hpp>
-
+#include <mongocxx/exception/exception.hpp>
 #include <mongocxx/stdx.hpp>
 
 namespace mongocxx {
 MONGOCXX_INLINE_NAMESPACE_BEGIN
-namespace exception {
-
-using error_and_code_type = std::tuple<std::string, std::int32_t>;
 
-class MONGOCXX_API base : virtual public std::exception {
+class MONGOCXX_API authentication_exception : public exception {
```

Comment:
Is there a reason you might want to catch operation exceptions and not authentication exceptions?